awswrangler/s3/_list.py

```diff
@@ -86,13 +89,45 @@ def _list_objects(  # pylint: disable=too-many-branches
     last_modified_begin: Optional[datetime.datetime] = None,
     last_modified_end: Optional[datetime.datetime] = None,
     ignore_empty: bool = False,
+    s3_list_strategy: Optional[str] = None,
```

Comment:
Shouldn't we add this parameter and `apply_config` on `list_objects` rather than this private function. That way customers will be able to see the parameter, and the docs will state that the parameter can be controlled by the config.

Reply:
That was my original thought but then I realized that you would have to apply this to ALL methods that use list_objects under the hood (delete_objects, describe_objects, list_objects, read_parquet, read_parquet_table, read_parquet_metadata, read_csv, read_json, read_fwf and many more) which seemed like an overkill for such a small edge case

Reply:
Yikes, I didn't realize there were that many. Is there a downside for the customer though if we add this parameter to all these functions? My main concern is still that this parameter will be hidden from customers. If it's applied on all the functions, it'll be clear which ones support it and which ones don't.